### PR TITLE
Add image support to Pillar component

### PR DESCRIPTION
**Pull request by danielguillan**
## Summary

Adds support to include an image as part of the Pillar component.

## List of notable changes:

- Added `<Pillar.Image>`.
- Updated interface guidelines and React docs.
- Updated the bottom margin of images to ensure consistency across Pillars and Cards.

## Steps to test:

- [Storybook](https://primer-68118d3fd5-26139705.drafts.github.io/brand/storybook/?path=/story/components-pillar-features--with-image)
- [Interface guidelines](https://primer-68118d3fd5-26139705.drafts.github.io/brand/components/Pillar)
- [React docs](https://primer-68118d3fd5-26139705.drafts.github.io/brand/components/Pillar/react) 

## Supporting resources (related issues, external links, etc):

- Resolves https://github.com/primer/brand/issues/836

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [x] UI Changes contain new visual snapshots (generated by adding `update snapshots` label to the PR)
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

**Comment by natalie-iv** [comment]
@jesussandreas could you please take a look. we had [that discussion in Slack](https://github.slack.com/archives/C04E44ZPNEA/p1733917322796519) around margin below the image. Would love to hear your perspective.

File: apps/docs/content/components/Pillar/react.mdx [hunk 1]
```diff
@@ -49,6 +49,25 @@ You can add an `icon` to enhance the visual context. We recommend using an [Octi
...
+### Image
+
+You can add an `image` to enhance the visual context.
+
+```jsx live
+<Pillar>
+  <Pillar.Image
+    src="https://via.placeholder.com/600x400/d3d9df/d3d9df.png"
+    alt="placeholder, blank area with an gray background color"
+    aspectRatio="16:9"
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/react.mdx:61]: Could we keep these examples minimal for simpler copy/pasting, or do you need the aspectRatio here?

```suggestion
```

File: packages/react/src/Pillar/Pillar.features.stories.tsx [hunk 2]
```diff
@@ -21,6 +22,22 @@ export const Icon: StoryFn<typeof Pillar> = () => {
   )
 }
 
+export const Image: StoryFn<typeof Pillar> = () => {
+  return (
+    <Pillar>
+      <Pillar.Image
+        aspectRatio="16:10"
```
Comment (rezrah) [inline packages/react/src/Pillar/Pillar.features.stories.tsx:29]: Same as aspectRatio

```suggestion
```

File: packages/react/src/Pillar/Pillar.module.css [hunk 3]
```diff
@@ -15,6 +15,17 @@
   text-align: left;
 }
 
+.Pillar__image {
+  margin-bottom: var(--base-size-28);
+  border-radius: var(--brand-borderRadius-medium);
```
Comment (rezrah) [inline packages/react/src/Pillar/Pillar.module.css:20]: All border-radius for cards, bento's etc are supposedly increasing to xlarge. Can we apply that now? Maybe also sneak in the same change into Card too?

File: .changeset/proud-students-collect.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react-brand': patch
+---
+
+Added image support to Pillar component
```
Comment (rezrah) [inline .changeset/proud-students-collect.md:5]: For easier scanning in the Changelog...

```suggestion
Added image support to `Pillar` component
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 5]
```diff
@@ -17,34 +20,35 @@ export default ComponentLayout
 
 ## Usage
 
-Use pillars to group similar features or benefits together to easily scan and read. Pillars display content in a familiar and recognizable style.
+Use pillars to display a collection of features or benefits, making them easier to scan and read. Pillars present content in a familiar and recognizable style.
 
 ### Stacked
 
-Pillars should be stacked to display a list of items that are related to each other. Use consistent pillar sizes within a row or group. A group or a row of content pillars is more visually appealing when the widths and heights of all elements match, so keep the content lenght of each pillar consistent.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:24]: 🙈 thanks for tidying up those typo's.. wondering if we should enable a CI spell checker

File: apps/docs/content/components/Pillar/index.mdx [hunk 6]
```diff
@@ -17,34 +20,35 @@ export default ComponentLayout
...
+Stack multiple pillars to create a collection of items. Ensure that all the items in a group have consistent content lengths so that their width and height match, making the content more visually appealing. When displaying a visual, create a balanced layout by using the same type for each pillar, either an icon or an image.
 
 <DoDontContainer>
   <Do>
     <img src="https://github.com/primer/brand/assets/912236/cd9679ab-cfa0-4c6d-8a55-21d16668f1ca" />
     <Caption>
-      Use a similar content size to achieve a visually balanced layout.
+      Use similar content lengths to achieve a visually balanced layout.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:33]: Nit... 'content lengths' feels a bit abstract. Could we call it character count to make it more obvious?

```suggestion
      Use similar character counts to achieve a visually balanced layout.
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 7]
```diff
@@ -17,34 +20,35 @@ export default ComponentLayout
...
     <img src="https://github.com/primer/brand/assets/912236/cd6e471f-98cf-4c70-801f-c0aeb4d2bdf5" />
     <Caption>
-      Don't use different length sizes for the title or the description.
+      Avoid varying content lengths for the title and description.
     </Caption>
   </Dont>
 </DoDontContainer>
 
-If displaying an icon, do so for all cards to achieve a visually balanced layout.
+Use pillars with images only for groups of a maximum of three to six items. Avoid stacking multiple sections that each contain groups of six pillars.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:44]: Recommend drawing more attention to this by wrapping in a warning Note. I think it's a little lost right now. Could also benefit from a visual do/dont example if you can do it.

```suggestion
<Note variant="warning">
Use pillars with images only for groups of a maximum of three to six items. Avoid stacking multiple sections that each contain groups of six pillars.
</Note>
```

File: apps/docs/content/components/Pillar/react.mdx (hunk 1 above)
Comment (danielguillan) [reply apps/docs/content/components/Pillar/react.mdx:61]: I'm just replicating what we currently have for the Card examples. I can remove it on both to make the examples minimal.

File: packages/react/src/Pillar/Pillar.module.css (hunk 3 above)
Comment (danielguillan) [reply packages/react/src/Pillar/Pillar.module.css:20]: I prefer that we handle this in a dedicated PR to ensure all components that need updating are addressed.

File: apps/docs/content/components/Pillar/index.mdx (hunk 5 above)
Comment (danielguillan) [reply apps/docs/content/components/Pillar/index.mdx:24]: That would be neat!

File: apps/docs/content/components/Pillar/index.mdx [hunk 8]
```diff
@@ -71,18 +89,18 @@ Cards should be used for noteworthy information to immediately draw user attenti
 
 #### Group size
 
-Cards should be used in small groups (3, 4 or 5 items maximum), whereas Pillars can be used in larger (more than 5 items) or smaller groups. As users encounter an increasing number of items within a group, the distinctiveness of each individual element diminishes.
+Cards should be used in small groups of three to five items, while Pillars can accommodate larger groups of up to six items. The more items in a group, the less distinctive each individual element becomes.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:94]: I thought the six Pillar limit was only for Pillars with images? AFAIK we should still be able to do up to 9 without  (3x3 grid on desktop).

File: apps/docs/content/components/Pillar/react.mdx [hunk 9]
```diff
@@ -49,6 +49,25 @@ You can add an `icon` to enhance the visual context. We recommend using an [Octi
 </Pillar>
 ```
 
+### Image
+
+You can add an `image` to enhance the visual context.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/react.mdx:54]: Don't think this needs inline code block here...

```suggestion
You can add an image to enhance the visual context.
```

File: apps/docs/content/components/Pillar/index.mdx (hunk 8 above)
Comment (danielguillan) [reply apps/docs/content/components/Pillar/index.mdx:94]: You're right, good catch!

File: packages/react/src/recipes/FeaturePreviewLPs/FeaturePreviewLevelZero/FeaturePreviewLevelZero.visual.spec.ts-snapshots/Visual-Comparison-FeaturePreviewLevelZero-FeaturePreviewLevelZero-Level-0-1-linux.png
Comment (rezrah) [inline packages/react/src/recipes/FeaturePreviewLPs/FeaturePreviewLevelZero/FeaturePreviewLevelZero.visual.spec.ts-snapshots/Visual-Comparison-FeaturePreviewLevelZero-FeaturePreviewLevelZero-Level-0-1-linux.png:1]: Remote assets aren't loading in, causing tests to incorrectly regen. Likely related to ongoing incident, so let's reset with `main` and then regen fresh tommorrow.

File: packages/react/src/Pillar/Pillar.test.tsx [hunk 10]
```diff
@@ -83,6 +83,23 @@ describe('Pillar', () => {
     expect(pillarEl).toHaveClass(classToCheck)
   })
 
+  it('renders the image correctly into the document', () => {
+    const mockTestId = 'test'
```
Comment (rezrah) [inline packages/react/src/Pillar/Pillar.test.tsx:87]:
```suggestion
    const mockTestId = 'test'
    const mockImage = 'mock.png'
```

File: packages/react/src/Pillar/Pillar.test.tsx [hunk 11]
```diff
@@ -83,6 +83,23 @@ describe('Pillar', () => {
...
+    const classToCheck = 'Image'
+    const testAltText = 'alternative text'
+
+    const {getByAltText} = render(
+      <Pillar data-testid={mockTestId}>
+        <Pillar.Image src="https://via.placeholder.com/600x400/d3d9df/d3d9df.png" alt={testAltText} />
```
Comment (rezrah) [inline packages/react/src/Pillar/Pillar.test.tsx:93]:
```suggestion
        <Pillar.Image src={mockImage} alt={testAltText} />
```

File: packages/react/src/Pillar/Pillar.module.css (hunk 3 above)
Comment (rezrah) [reply packages/react/src/Pillar/Pillar.module.css:20]: Okay, mentioning as I can see [you've made a padding update to the Card in this PR](https://github.com/primer/brand/pull/862/files#diff-666b3aa4d4d292698c805bf515a4cd8ff79c3d7f4db003e28aaf4efe81bde8c8R103) already, so felt okay to do it in this PR too? Fine if you want to defer it also.

**Review by rezrah — APPROVED** [review]

**Pull request merged**